lib/evm/opcodes.ts

```diff
@@ -1,16 +1,41 @@
 import Common from 'ethereumjs-common'
 
-export interface Opcode {
-  name: string
-  fee: number
-  isAsync: boolean
+export class Opcode {
+  readonly code: number
+  readonly name: string
+  readonly fullName: string
+  readonly fee: number
+  readonly isAsync: boolean
+
+  constructor({
+    code,
+    name,
+    fullName,
+    fee,
+    isAsync,
+  }: {
+    code: number
+    name: string
+    fullName: string
+    fee: number
+    isAsync: boolean
+  }) {
+    this.code = code
+    this.name = name
+    this.fullName = fullName
+    this.fee = fee
+    this.isAsync = isAsync
+
+    // Opcode isn't subject to change, thus all futher modifications are prevented.
+    Object.freeze(this)
```

Comment:
This is one thing that's slightly unusual and I wonder if anyone else has a view about it...

It's providing an extra margin of safety so no one can accidentally modify it via the `step`? 

There are some small implications for anyone who might be cloning or copying things in the step - frozenness is inherited in some cases (see [Creating new objects from frozen parent objects][1])

[1]:

Reply:
Example on StackOverflow has one non-obvious effect. `Object.create()` produces a link in the memory to the source object. I'd suggest to copy properties with spread operator:

```js
const opcode = new Opcode({
  ...source,
 fee: source.fee + 10
})
```

Reply:
👍